image_classification.py

```diff
@@ -0,0 +1,559 @@
+                                                  workers=1, use_multiprocessing=False,
+                                                  verbose=1)
+            def flatten(y):
+                return np.argmax(y, axis=1).reshape(-1)
+            
+            score = cohen_kappa_score(flatten(self.y_val),
+                                      flatten(y_pred),
+                                      labels=[0,1,2,3,4],
+                                      weights='quadratic')
+            print("\n epoch: %d - QWK_score: %.6f \n" % (epoch+1, score))
+            self.history.append(score)
+            if score >= max(self.history):
+                print('saving checkpoint: ', score)
+                self.model.save('../working/densenet_bestqwk.h5')
+
+qwk = QWKEvaluation(validation_data=(valid_generator, valid_y),
+                    batch_size=batch_size, interval=1)
+
+
+# In[16]:
+
+# warm up model
+for layer in model.layers:
+    layer.trainable = False
+
+for i in range(-3,0):
+    model.layers[i].trainable = True
+
+model.compile(
+    loss='categorical_crossentropy',
+    optimizer=Adam(1e-3))
+
+model.fit_generator(
+    train_generator,
+    steps_per_epoch=np.ceil(float(len(train_y)) / float(128)),
+    epochs=2,
+    workers=WORKERS, use_multiprocessing=True,
+    verbose=1,
+    callbacks=[qwk])
+
+
+# In[17]:
+
+# train all layers
+for layer in model.layers:
+    layer.trainable = True
+callbacks_list = [checkpoint, csv_logger, reduceLROnPlat, early, qwk]
+model.compile(loss='categorical_crossentropy',
+            # loss=kappa_loss,
+            optimizer=Adam(lr=1e-4))
+model.fit_generator(
+    train_mixup,
+    steps_per_epoch=np.ceil(float(len(train_x)) / float(batch_size)),
+    validation_data=valid_generator,
+    validation_steps=np.ceil(float(len(valid_x)) / float(batch_size)),
+    epochs=epochs,
+    verbose=1,
+    workers=1, use_multiprocessing=False,
+    callbacks=callbacks_list)
+
+
+# In[18]:
+
+submit = pd.read_csv('../input/aptos2019-blindness-detection/sample_submission.csv')
+model.load_weights('../working/densenet_bestqwk.h5')
+predicted = []
+
+
+# In[19]:
+
+# reference:https://www.kaggle.com/CVxTz/cnn-starter-nasnet-mobile-0-9709-lb 
+for i, name in tqdm(enumerate(submit['id_code'])):
+    path = os.path.join('../input/aptos2019-blindness-detection/test_images/', name+'.png')
+    image = cv2.imread(path)
+    image = cv2.resize(image, (SIZE, SIZE))
+    image = cv2.addWeighted ( image,4, cv2.GaussianBlur( image , (0,0) , 10) ,-4 ,128)
+    X = np.array((image[np.newaxis])/255)
+    score_predict=((model.predict(X).ravel()*model.predict(X[:, ::-1, :, :]).ravel()*model.predict(X[:, ::-1, ::-1, :]).ravel()*model.predict(X[:, :, ::-1, :]).ravel())**0.25).tolist()
+    label_predict = np.argmax(score_predict)
+    predicted.append(str(label_predict))
+
+
+# In[20]:
+
+submit['diagnosis'] = predicted
+submit.to_csv('submission.csv', index=False)
+submit.head()
+
```

Comment:
@zhangabner Lines 533-559 (image_classification.py) are different from Lines 458-464 (tasks/computer-vision/image-classification/image_classification.py). Was this intentional?

Reply:
Yes, I use mlflow to track the training, not to inference yet.